### PR TITLE
Monorepo setup

### PR DESCRIPTION
**Pull request by konraddysput**
# Why

This diff adds monorepo basic setup to the repository.

### CODE REVIEW

File: packages/core/src/BacktraceCoreClient.ts [hunk 1]
```diff
@@ -0,0 +1,8 @@
+import { BacktraceAttachment } from './model/report/BacktraceAttachment';
+import { BacktraceReport } from './model/report/BacktraceReport';
+
+export class BacktraceCoreClient {
+    public async send(report: BacktraceReport, attributes: Record<string, any>, attachments: BacktraceAttachment) {
```
Comment (perf2711) [inline packages/core/src/BacktraceCoreClient.ts:5]: why `attributes` and `attachments` are not part of `BacktraceReport`?

File: packages/core/src/model/configuration/BacktraceConfiguration.ts [hunk 2]
```diff
@@ -0,0 +1,51 @@
+import { BacktraceAttachment } from '../report/BacktraceAttachment';
+import { BacktraceDatabaseConfiguration } from './BacktraceDatabaseConfiguration';
+
+// Submission information
+export type BacktraceBasicSubmissionInformation = {
+    /**
+     * The server address (submission URL) is required to submit exceptions from your Unity project to your Backtrace instance.
```
Comment (perf2711) [inline packages/core/src/model/configuration/BacktraceConfiguration.ts:7]: >Unity

I think you missed this part

File: packages/core/src/BacktraceCoreClientBuilder.ts [hunk 3]
```diff
@@ -0,0 +1,8 @@
+import { BacktraceCoreClient } from './BacktraceCoreClient';
+import { BacktraceConfiguration } from './model/configuration/BacktraceConfiguration';
+
+export class BacktraceCoreClientBuilder {
+    public static build(options?: BacktraceConfiguration): BacktraceCoreClient {
+        throw new Error('Not implemented');
+    }
+}
```
Comment (perf2711) [inline packages/core/src/BacktraceCoreClientBuilder.ts:8]: Will we ever build core client, if other clients will derive from it?

File: packages/core/src/BacktraceCoreClient.ts (hunk 1 above)
Comment (konraddysput) [reply packages/core/src/BacktraceCoreClient.ts]: Done. Thx

File: packages/core/src/model/configuration/BacktraceConfiguration.ts (hunk 2 above)
Comment (konraddysput) [reply packages/core/src/model/configuration/BacktraceConfiguration.ts]: Done Thx

File: packages/sdk-core/src/model/report/BacktraceReport.ts [hunk 4]
```diff
@@ -0,0 +1,7 @@
+import { BacktraceAttachment } from './BacktraceAttachment';
+
+export interface BacktraceReport {
+    attributes(): Record<string, unknown>;
+    attachments(): BacktraceAttachment[];
+    toDate(): Record<string, unknown>;
```
Comment (perf2711) [inline packages/sdk-core/src/model/report/BacktraceReport.ts:6]: what is `toDate` exactly? Is there a reason that these are methods, and not `readonly` fields?

Comment (konraddysput) [reply packages/sdk-core/src/model/report/BacktraceReport.ts:6]: those are methods and generate something  - toDate convert report to the data object that we can send to Backtrace. Those information will more clean soon

Comment (konraddysput) [reply packages/sdk-core/src/model/report/BacktraceReport.ts]: Ups, I see - typo

File: packages/node/package.json [hunk 5]
```diff
@@ -0,0 +1,36 @@
+{
+    "name": "@backtrace/node",
+    "version": "0.0.1",
+    "description": "Backtrace-JavaScript Node.JS integration",
+    "main": "lib/index.js",
+    "types": "lib/index.d.ts",
+    "scripts": {
+        "test": "NODE_ENV=test jest"
```
Comment (perf2711) [inline packages/node/package.json:8]: we are missing `build`, `rebuild` etc. scripts here

File: packages/browser/package.json [hunk 6]
```diff
@@ -0,0 +1,32 @@
+{
+    "name": "@backtrace/browser",
+    "version": "0.0.1",
+    "description": "Backtrace-JavaScript web browser integration",
+    "main": "lib/index.js",
+    "types": "lib/index.d.ts",
+    "scripts": {
+        "test": "NODE_ENV=test jest"
```
Comment (perf2711) [inline packages/browser/package.json:8]: we are missing `build`, `rebuild` etc. scripts here

File: packages/sdk-core/package.json [hunk 7]
```diff
@@ -0,0 +1,32 @@
+{
+    "name": "@backtrace/sdk-core",
+    "version": "0.0.1",
+    "description": "Backtrace-JavaScript SDK core library",
+    "main": "lib/index.js",
+    "types": "lib/index.d.ts",
+    "scripts": {
+        "test": "NODE_ENV=test jest"
```
Comment (perf2711) [inline packages/sdk-core/package.json:8]: we are missing build, rebuild etc. scripts here

File: packages/sdk-core/src/model/configuration/BacktraceConfiguration.ts [hunk 8]
```diff
@@ -0,0 +1,51 @@
...
+     */
+    attachments?: BacktraceAttachment[];
+
+    /**
+     * Metrics such as crash free users and crash free sessions
+     */
+    metrics?: BacktraceMetricsSupport;
+};
+
+export type BacktraceConfiguration = BasicBacktraceClientSettings & BacktraceDatabaseConfiguration;
```
Comment (perf2711) [inline packages/sdk-core/src/model/configuration/BacktraceConfiguration.ts:51]: why are we not using `interface` here in these options (not the union ones)?

File: packages/sdk-core/src/model/configuration/BacktraceDatabaseConfiguration.ts [hunk 9]
```diff
@@ -0,0 +1,71 @@
+export enum DeduplicationStrategy {
+    None = 0,
+    Callstack = 1 << 0,
+    Classifier = 1 << 1,
+    Message = 1 << 2,
+    All = ~(~0 << 4),
+}
+export type EnabledBacktraceConfiguration = {
```
Comment (perf2711) [inline packages/sdk-core/src/model/configuration/BacktraceDatabaseConfiguration.ts:8]: why not `interface`?

File: packages/sdk-core/src/model/report/BacktraceAttachment.ts [hunk 10]
```diff
@@ -0,0 +1,13 @@
+export type BacktraceInMemoryAttachment = {
```
Comment (perf2711) [inline packages/sdk-core/src/model/report/BacktraceAttachment.ts:1]: why are we not using `interface` here in these options?

File: packages/node/package.json (hunk 5 above)
Comment (konraddysput) [reply packages/node/package.json:13]: done

File: packages/browser/package.json (hunk 6 above)
Comment (konraddysput) [reply packages/browser/package.json:13]: done

File: packages/sdk-core/package.json (hunk 7 above)
Comment (konraddysput) [reply packages/sdk-core/package.json:13]: done

File: packages/sdk-core/src/model/configuration/BacktraceConfiguration.ts (hunk 8 above)
Comment (konraddysput) [reply packages/sdk-core/src/model/configuration/BacktraceConfiguration.ts:51]: why should we? Do we want to inherit from them? Is there any better reason to use interface than type?

Comment (perf2711) [reply packages/sdk-core/src/model/configuration/BacktraceConfiguration.ts:51]: it seems more suitable to use an interface to define an object contract. I would use `type` to define a type alias, like  `type BacktraceConfiguration = BasicBacktraceClientSettings & BacktraceDatabaseConfiguration;`, or `type Key = string | number`.

Comment (konraddysput) [reply packages/sdk-core/src/model/configuration/BacktraceConfiguration.ts]: Pushed adjusted code base on our conversation

File: packages/sdk-core/src/model/configuration/BacktraceDatabaseConfiguration.ts (hunk 9 above)
Comment (konraddysput) [reply packages/sdk-core/src/model/configuration/BacktraceDatabaseConfiguration.ts]: Pushed adjusted code base on our conversation

File: packages/sdk-core/src/model/report/BacktraceAttachment.ts (hunk 10 above)
Comment (konraddysput) [reply packages/sdk-core/src/model/report/BacktraceAttachment.ts]: Pushed adjusted code base on our conversation

**Review by perf2711 — APPROVED** [review]

**Pull request merged**